### PR TITLE
ZCS-12676 WCAG | Classic UI | High Complexity | Tinymce issues

### PR DESCRIPTION
**Pull request by miteshsavani810**
**3.3.2_label issue-New Mail.docx**
> Fixed: Source code dialog doesn't have aria-label associate to textbox hence Screen reader is not announcing it properly.

**2.1.1_access issue-New Mail Dialog.docx** 
> Fixed: Combobox element not accessible by keyboard navigation using Tab key.
> Fixed: Emoticon dialog's icons are not accessible by keyboard. After fix, user can access all icons of emoticon panel with Arrow keys.

### CODE REVIEW

File: tiny_mce/custom/plugins/zemoticons/plugin.js [hunk 1]
```diff
@@ -71,6 +71,27 @@ tinymce.PluginManager.add('zemoticons', function(editor, url) {
         return emoticonsHtml.join('');
     }
 
+    function _keyDown(ev) {
+
+        const icons = (Array.from(this.querySelectorAll('a')));
```
Comment (silentsakky) [inline tiny_mce/custom/plugins/zemoticons/plugin.js:76]:
```suggestion
        const icons = Array.from(this.querySelectorAll('a'));
```

File: tiny_mce/custom/plugins/zemoticons/plugin.js [hunk 2]
```diff
@@ -71,6 +71,27 @@ tinymce.PluginManager.add('zemoticons', function(editor, url) {
...
+        const totalIcons = icons.length;
+        const currentIconIndex = icons.indexOf(ev.target);
+        let nextIconIndex = null;
+
+        if (ev.keyCode === 37 || ev.keyCode === 39) {
+            nextIconIndex = ev.keyCode === 37 ? totalIcons + currentIconIndex - 1 : currentIconIndex + 1; 
+        }
+        else if (ev.keyCode === 38 || ev.keyCode === 40) {
```
Comment (silentsakky) [inline tiny_mce/custom/plugins/zemoticons/plugin.js:84]:
```suggestion
        } else if (ev.keyCode === 38 || ev.keyCode === 40) {
```

File: tiny_mce/custom/plugins/zemoticons/plugin.js [hunk 3]
```diff
@@ -71,6 +71,27 @@ tinymce.PluginManager.add('zemoticons', function(editor, url) {
...
+            nextIconIndex = ev.keyCode === 38 ? totalIcons + currentIconIndex - 4 : currentIconIndex + 4;
+        }
+
+        if (nextIconIndex) {
+            const nextIcon = icons[nextIconIndex%totalIcons];
```
Comment (silentsakky) [inline tiny_mce/custom/plugins/zemoticons/plugin.js:89]: let/const not used in classic ui

Comment (silentsakky) [inline tiny_mce/custom/plugins/zemoticons/plugin.js:89]:
```suggestion
            var nextIcon = icons[nextIconIndex % totalIcons];
```

File: tiny_mce/custom/plugins/zemoticons/plugin.js [hunk 4]
```diff
@@ -88,6 +109,15 @@ tinymce.PluginManager.add('zemoticons', function(editor, url) {
             }
         },
         icon: 'emoticons',
-        tooltip: 'Emoticons'
+        tooltip: 'Emoticons',
+        onclick: function (e) {
+            var visiblePanel = Array.from(document.querySelectorAll('table[class="mce-grid"]')).filter((el) => el.parentElement.style.display === '');
```
Comment (silentsakky) [inline tiny_mce/custom/plugins/zemoticons/plugin.js:114]: no arrow function

File: tiny_mce/custom/plugins/zemoticons/plugin.js [hunk 5]
```diff
@@ -88,6 +109,15 @@ tinymce.PluginManager.add('zemoticons', function(editor, url) {
...
+            if (visiblePanel.length){
```
Comment (silentsakky) [inline tiny_mce/custom/plugins/zemoticons/plugin.js:115]:
```suggestion
            if (visiblePanel.length) {
```

File: tiny_mce/custom/plugins/zemoticons/plugin.js (hunk 1 above)
Comment (miteshsavani810) [reply tiny_mce/custom/plugins/zemoticons/plugin.js]: Done.

File: tiny_mce/custom/plugins/zemoticons/plugin.js (hunk 2 above)
Comment (miteshsavani810) [reply tiny_mce/custom/plugins/zemoticons/plugin.js]: Done

File: tiny_mce/custom/plugins/zemoticons/plugin.js (hunk 3 above)
Comment (miteshsavani810) [reply tiny_mce/custom/plugins/zemoticons/plugin.js]: Fixed

Comment (miteshsavani810) [reply tiny_mce/custom/plugins/zemoticons/plugin.js]: Done

File: tiny_mce/custom/plugins/zemoticons/plugin.js (hunk 4 above)
Comment (miteshsavani810) [reply tiny_mce/custom/plugins/zemoticons/plugin.js]: Replaced

File: tiny_mce/custom/plugins/zemoticons/plugin.js (hunk 5 above)
Comment (miteshsavani810) [reply tiny_mce/custom/plugins/zemoticons/plugin.js]: Done

**Review by silentsakky — APPROVED** [review]

**Review by lavegupta — APPROVED** [review]

**Pull request merged**